sdk_extensions/zpages/src/main/java/io/opentelemetry/sdk/extensions/zpages/TracezZPageHandler.java

```diff
@@ -0,0 +1,594 @@
+
+    int getValue() {
+      return value;
+    }
+  }
+
+  private static final String TRACEZ_URL = "/tracez";
+  // Background color used for zebra striping rows of summary table
+  private static final String ZEBRA_STRIPE_COLOR = "#e6e6e6";
+  // Color for sampled traceIds
+  private static final String SAMPLED_TRACE_ID_COLOR = "#c1272d";
+  // Color for not sampled traceIds
+  private static final String NOT_SAMPLED_TRACE_ID_COLOR = "black";
+  // Query string parameter name for span name
+  private static final String PARAM_SPAN_NAME = "zspanname";
+  // Query string parameter name for type to display
+  // * 0 = running, 1 = latency, 2 = error
+  private static final String PARAM_SAMPLE_TYPE = "ztype";
+  // Query string parameter name for sub-type:
+  // * for latency based sampled spans [0, 8] corresponds to each latency boundaries
+  //   where 0 corresponds to the first boundary
+  // * for error based sampled spans [0, 15], 0 means all, otherwise the error code
+  private static final String PARAM_SAMPLE_SUB_TYPE = "zsubtype";
+  // Map from LatencyBoundaries to human readable string on the UI
+  private static final ImmutableMap<LatencyBoundaries, String> LATENCY_BOUNDARIES_STRING_MAP =
+      buildLatencyBoundariesStringMap();
+  @Nullable private final TracezDataAggregator dataAggregator;
+
+  /** Constructs a new {@code TracezZPageHandler}. */
+  TracezZPageHandler(@Nullable TracezDataAggregator dataAggregator) {
+    this.dataAggregator = dataAggregator;
+  }
+
+  @Override
+  public String getUrlPath() {
+    return TRACEZ_URL;
+  }
+
+  /**
+   * Emits CSS Styles to the {@link PrintStream} {@code out}. Content emitted by this function
+   * should be enclosed by <head></head> tag.
+   *
+   * @param out the {@link PrintStream} {@code out}.
+   */
+  private static void emitHtmlStyle(PrintStream out) {
+    out.print("<style>");
+    out.print(ZPageStyle.style);
+    out.print("</style>");
+  }
+
+  /**
+   * Emits the header of the summary table to the {@link PrintStream} {@code out}.
+   *
+   * @param out the {@link PrintStream} {@code out}.
+   * @param formatter a {@link Formatter} for formatting HTML expressions.
+   */
+  private static void emitSummaryTableHeader(PrintStream out, Formatter formatter) {
+    // First row
+    out.print("<tr class=\"bg-color\">");
+    out.print("<th colspan=1 class=\"header-text\"><b>Span Name</b></th>");
+    out.print("<th colspan=1 class=\"header-text border-left-white\"><b>Running</b></th>");
+    out.print("<th colspan=9 class=\"header-text border-left-white\"><b>Latency Samples</b></th>");
+    out.print("<th colspan=1 class=\"header-text border-left-white\"><b>Error Samples</b></th>");
+    out.print("</tr>");
+
+    // Second row
+    out.print("<tr class=\"bg-color\">");
+    out.print("<th colspan=1></th>");
+    out.print("<th colspan=1 class=\"border-left-white\"></th>");
+    for (LatencyBoundaries latencyBoundaries : LatencyBoundaries.values()) {
+      formatter.format(
```

Comment:
Since performance of this page isn't a big deal, it's not too bad but reconsider using `Formatter`, it's orders of magnitude slower than concatenation and while useful for formatting floating points, otherwise is not doing much more than concatenation. It's generally good practice to just never use `String.format` / `Formatter` and not worry about the performance tradeoff.

Reply:
So doing `out.print()` and concatenate the variables is a better choice if no formatting is required then?

Reply:
Yeah I would say it's a better default - otherwise you have to make a judgment call on performance vs readability, but I think the readability gain from printf is usually very low (interpolated strings would be a different story, eventually they'll be in Java and hopefully fast to make up for decades ;) )

Reply:
Got it, I will remove unnecessary `Formatter` and change them to concatenate